Src/FluentAssertions/Types/TypeSelector.cs

```diff
@@ -165,6 +167,96 @@ public TypeSelector ThatAreNotUnderNamespace(string @namespace)
             return this;
         }
 
+        /// <summary>
+        /// Determines whether the type is class
+        /// </summary>
+        public TypeSelector ThatAreClasses()
+        {
+            types = types.Where(t => t.IsClass).ToList();
+            return this;
+        }
+        /// <summary>
+        /// Determines whether the type is not a class
+        /// </summary>
+        public TypeSelector ThatAreNotClasses()
+        {
+            types = types.Where(t => !t.IsClass).ToList();
+            return this;
+        }
+
+        /// <summary>
+        /// Determines whether the type is a static class
+        /// </summary>
+        public TypeSelector ThatAreStaticClasses()
+        {
+            types = types.Where(t => t.IsCSharpStatic()).ToList();
+            return this;
+        }
+
+        /// <summary>
+        /// Determines whether the type is not a static class
+        /// </summary>
+        public TypeSelector ThatAreNotStaticClasses()
+        {
+            types = types.Where(t => !t.IsCSharpStatic()).ToList();
+            return this;
+        }
+
+        /// <summary>
+        /// Allows to filter the types with the <paramref name="predicate"/> passed
+        /// </summary>
+        public TypeSelector ThatAre(Func<Type, bool> predicate)
+        {
+            types = types.Where(predicate).ToList();
+            return this;
+        }
+
+        /// <summary>
+        /// Returns T for the types which are Task&lt;T&gt;; the type itself otherwise
+        /// </summary>
+        public TypeSelector UnwrapTaskTypes()
+        {
+            types = types.Select(type =>
+            {
+                if (type.IsGenericType && type.GetGenericTypeDefinition() == typeof(Task<>))
+                {
+                    return type.GetGenericArguments().Single();
+                }
+                return type == typeof(Task) ? typeof(void) : type;
+            }).ToList();
+
+            return this;
+        }
+
+        /// <summary>
+        /// Returns T for the types which are IEnumerable&lt;T&gt; or implement the IEnumerable&lt;T&gt;; the type itself otherwise
+        /// </summary>
+        public TypeSelector UnwrapEnumerableTypes()
+        {
+            types = types
+                .Select(type =>
+                {
+                    if (type.IsGenericType)
+                    {
+                        if (type.GetGenericTypeDefinition() == typeof(IEnumerable<>))
+                        {
+                            return type.GetGenericArguments().Single();
+                        }
+
+                        var iEnumerableDefinition = type.GetInterfaces()
+                            .SingleOrDefault(iType => iType.IsGenericType && iType.GetGenericTypeDefinition() == typeof(IEnumerable<>));
```

Comment:
This might make too strong assumptions about the user's code.

E.g. if a type has multiple implementations of `IEnumerable<>` this throws
```c#
public class C : IEnumerable<int>, IEnumerable<string>
{
    private readonly IEnumerable<int> integers;
    private readonly IEnumerable<string> strings;

    public IEnumerator<int> GetEnumerator() => integers.GetEnumerator();
    
    IEnumerator IEnumerable.GetEnumerator() => ((IEnumerable)integers).GetEnumerator();
    
    IEnumerator<string> IEnumerable<string>.GetEnumerator() => strings.GetEnumerator();
}
```

```
System.InvalidOperationException: 'Sequence contains more than one matching element'
```

Reply:
Done